amundsen_application/static/js/components/common/Inputs/CheckBoxItem/index.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import './styles.scss';
+
+export interface CheckBoxItemProps {
+  checked: boolean;
+  disabled?: boolean
+  name: string;
+  onChange: (e: React.FormEvent<HTMLInputElement>) => any;
```

Comment:
Might be worthwhile to abstract away the actual click event since most users of a checkbox are not interested in the event itself.
```
<input onChange={ (e) => this.props.onChange(e.target.value) }
```

Reply:
Unless we have reason to, this pattern is something that is discouraged due the "jsx-no-lambda rule". https://jonhilton.net/typescript-and-react-forbidden-lambdas/

My understanding is that this is not a "bad" thing to do, but rather that they should be used more thoughtfully and perhaps only when needed. For example I know I've used it in `OwnerEditor` because of a scoping problem. Do you have any strong feelings or thoughts on this before this is merged?

Reply:
I used the lambda for convenience. I should have written out the full example as below.

https://github.com/lyft/amundsenfrontendlibrary/blob/c9ba2b878fbc7e10d5e991d63324a14c62291d7c/amundsen_application/static/js/components/SearchPage/ResourceSelector/index.tsx#L35-L49

Maybe this works differently for `React.SFC` since you can't define class functions. Don't have very strong feelings about this, but I think we should aim to make reusable components as simple to use as possible.

Reply:
I see. I think one major factor here does involve the fact that this is a `React.SFC`. The properties given will be placed directly on the `input` element, and TypeScript requires that this method match the expected web api.

Beyond that, I also see the purpose of this kind of component to be creating something that we can reuse anywhere in our application for current and future use cases. Therefore I believe we should design first for all use cases and second for ease of use -- but also weigh the use case coverage against how much more difficult it makes the component to use. In this case, I think designing for all use cases is preferable since it doesn't add any burden beyond having to grab `event.target.value` instead of `value`.

However I think the concept you raised is always worth keeping in mind as we create these components -- if it were the case that we created something that would require doing some kind of pre-processing (example: something that needs to sanitize the data in some way before using it), then that common component should get upgraded to a `React.Component` and have some helper method in the way that you described.